### PR TITLE
fix(bug): Return unknown when not authorized for watch status

### PR DESCRIPTION
**Pull request by connorgorman**
## Description

We should not block access to images if a user is not authorized to look at a watched image. Instead just return unknown.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

Will run with a role without watched image and ensure that the ui and everything looks good

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

**Review by theencee — APPROVED** [review]

File: ui/apps/platform/src/Components/CVEStackedPill/CVEStackedPill.js [hunk 1]
```diff
@@ -52,7 +52,7 @@ const CVEStackedPill = ({
     return (
         <div className="flex items-center w-full">
             {useScan && !hasScan && <span>{entityName} not scanned</span>}
-            {!hasCounts && <span>No CVEs</span>}
+            {(!useScan || hasScan) && !hasCounts && <span>No CVEs</span>}
```
Comment (vjwilson) [inline ui/apps/platform/src/Components/CVEStackedPill/CVEStackedPill.js:55]: Why is are the extra conditions necessary?

Expanded to what those flag vars are set by, the logic is
```suggestion
            {( !( !!scanTime) || scanTime !== '-') && !hasCounts && <span>No CVEs</span>}
            
            // which reduces to
           {( !scanTime || scanTime !== '-') && !hasCounts && <span>No CVEs</span>}
           
           // where scanTime is either `false`, or anything but '-'
```

I'm just trying to understand what will be returned for `scanTime`, if you don't have the WatchImages permission, and does it only affect Watch Images?

**Review by vjwilson — COMMENTED** [review]
I'd like to know how the backend change affects the `scanTime` response before I agree this is an appropriate FE change.

File: ui/apps/platform/src/Components/CVEStackedPill/CVEStackedPill.js (hunk 1 above)
Comment (connorgorman) [reply ui/apps/platform/src/Components/CVEStackedPill/CVEStackedPill.js:55]: Tbh this is a completely unrelated change. which fixes

Screenshot 2023-10-03 at 4 27 31 PM

Comment (connorgorman) [reply ui/apps/platform/src/Components/CVEStackedPill/CVEStackedPill.js:55]: I'll take this out of this PR and into a different PR

**Review by vjwilson — APPROVED** [review]

**Pull request merged**